pkg/util/detector/detector.go

```diff
@@ -528,15 +552,62 @@ func (d *ResourceDetector) BuildClusterResourceBinding(object *unstructured.Unst
 		},
 		Spec: workv1alpha1.ResourceBindingSpec{
 			Resource: workv1alpha1.ObjectReference{
-				APIVersion:      object.GetAPIVersion(),
-				Kind:            object.GetKind(),
-				Name:            object.GetName(),
-				ResourceVersion: object.GetResourceVersion(),
+				APIVersion:                  object.GetAPIVersion(),
+				Kind:                        object.GetKind(),
+				Name:                        object.GetName(),
+				ResourceVersion:             object.GetResourceVersion(),
+				ReplicaResourceRequirements: replicaResourceRequirements,
+				Replicas:                    replicas,
 			},
 		},
 	}
 
-	return binding
+	return binding, nil
+}
+
+// GetObjectResource get the replicas and resources of a object
+func (d *ResourceDetector) GetObjectResource(object *unstructured.Unstructured) (corev1.ResourceList, int32, error) {
+	if object.GetKind() == util.DeploymentKind {
```

Comment:
Does this support all deployments versions, such as v1beta1/v1/v1beta2?

Reply:
I am not sure. But this judgment already exists in elsewhere。
![image](https://user-images.githubusercontent.com/8034855/124537629-dc4f5700-de4c-11eb-88bf-63233ed3a709.png)